segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/S3ClientWrapper.java

```diff
@@ -0,0 +1,264 @@
+    public AccessControlList getObjectAcl(String bucketName, String key) {
+        AclSize retVal = aclMap.get(key);
+        if (retVal == null) {
+            throw new S3Exception("NoObject", 500, "NoSuchKey", key);
+        }
+        return retVal.getAcl();
+    }
+
+    @Override
+    public void deleteObject(String bucketName, String key) {
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        try {
+            Files.delete(path);
+        } catch (IOException e) {
+            throw new S3Exception("NoSuchKey", 0, "NoSuchKey", "");
+        }
+        aclMap.remove(key);
+    }
+
+    @Override
+    public ListObjectsResult listObjects(String bucketName, String prefix) {
+        ListObjectsResult result = new ListObjectsResult();
+        ArrayList<S3Object> list = new ArrayList<>();
+        Path path = Paths.get(this.baseDir, bucketName, prefix);
+        try {
+            if (Files.exists(path)) {
+                Files.list(path).forEach((file) -> {
+                    S3Object object = new S3Object();
+                    object.setKey(file.toString().replaceFirst(Paths.get(this.baseDir, bucketName).toString(), ""));
+                    list.add(object);
+                });
+            }
+        } catch (IOException e) {
+            throw new S3Exception("NoSuchKey", 0, "NoSuchKey", "");
+        }
+        result.setObjects(list);
+        return result;
+    }
+
+    @Override
+    public S3ObjectMetadata getObjectMetadata(String bucketName, String key) {
+        S3ObjectMetadata metadata = new S3ObjectMetadata();
+        AclSize data = aclMap.get(key);
+        if (data == null) {
+            throw new S3Exception("NoSuchKey", 0, "NoSuchKey", "");
+        }
+        metadata.setContentLength(data.getSize());
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        metadata.setLastModified(new Date(path.toFile().lastModified()));
+        return metadata;
+    }
+
+    @Override
+    public InputStream readObjectStream(String bucketName, String key, Range range) {
+        byte[] bytes = new byte[Math.toIntExact(range.getLast() + 1 - range.getFirst())];
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        FileInputStream returnStream;
+        try {
+            returnStream = new FileInputStream(path.toFile());
+            if ( range.getFirst() != 0) {
+                long bytesSkipped = 0;
+                do {
+                    bytesSkipped += returnStream.skip(range.getFirst());
+                } while (bytesSkipped < range.getFirst());
+            }
+            StreamHelpers.readAll(returnStream, bytes, 0, bytes.length);
+            return new ByteArrayInputStream(bytes);
+        } catch (IOException e) {
+            throw new S3Exception("NoSuchKey", 0, "NoSuchKey", "");
+        }
+    }
+
+    @Override
+    public String initiateMultipartUpload(String bucketName, String key) {
+        multipartUploads.put(key, new ConcurrentHashMap<>());
```

Comment:
I see things added to this map, but I don't ever see them removed. I think this is a memory leak.